doc/dev/mdbook.md

```diff
@@ -14,30 +23,94 @@ We are using the following settings while editing the markdown docs.
 
 ## Build
 
-This manual is generated by
-[mdBook](https://github.com/rust-lang/mdBook). We are currently using
-a [fork](https://github.com/leanprover/mdBook) of it for the following
-additional features:
+To build and test the book you have to preprocess the .lean files with Alectryon then use our own
+fork of the Rust tool named [mdbook](https://github.com/leanprover/mdbook). We have our own fork of
+mdBook with the following additional features:
 
 * Add support for hiding lines in other languages
   [#1339](https://github.com/rust-lang/mdBook/pull/1339)
-* Replace calling `rustdoc --test` from `mdbook test` with `./test`
+* Make `mdbook test` call the `lean` compiler to test the snippets.
+* Ability to test a single chapter at a time which is handy when you
+are working on that chapter.  See the `--chapter` option.
 
-To build this manual, first install the fork via
-```bash
-cargo install --git https://github.com/leanprover/mdBook mdbook
-```
-Then use e.g. [`mdbook watch`](https://rust-lang.github.io/mdBook/cli/watch.html) in the `doc/` folder:
+So you need to setup these tools before you can run `mdBook`.
 
-```bash
-cd doc
-mdbook watch --open  # opens the output in `out/` in your default browser
-```
+1. install [Rust](https://www.rust-lang.org/tools/install)
+which provides you with the `cargo` tool for building rust packages.
+Then run the following:
+    ```bash
+    cargo install --git https://github.com/leanprover/mdBook mdbook
+    ```
+
+1. Clone https://github.com/leanprover/LeanInk.git and run `lake build` then copy the resulting
+executable to your `$HOME/.elan/bin` folder or `%USERPROFILE%\.elan\bin` so Alectryon can find it
+there.
+
+1. Create a Python 3.10 environment, if you use anaconda then that would be:
+    ```
+    conda create -n leanink python=3.10
+    ```
+
+1. Activate this environment and install the following packages:
+    ```
+    pip install pygments dominate beautifulsoup4 docutils
+    ```
+
+1. Clone https://github.com/Kha/alectryon.git and run the following install the
+Alectryon python package:
+    ```
+    git checkout typeid
+    python setup.py install
+    ```
```

Comment:
This is pretty complicated, what's wrong with the `pip` one-liner from the Alectryon docs?

Reply:
What's complicated? The anaconda info?  The pip line is one line there?  It is important to mention it requires python 3.10 and that is usually easiest to do in a virtual environment, since it is not the default version on Ubuntu yet is it? (and what if they are not on Ubuntu?)  The git clone of and install alectryon is fine right?

Reply:
Hard-coding the dependencies of a third-party project in our docs is just wrong. You make a good point about the environment (though I'd prefer to see that point in Alectryon's own docs), but even then would
```
python3 -m pip install git+https://github.com/Kha/alectryon.git@typeid
```
not work?

Reply:
Much better, thanks, I updated the instructions.